usr/src/uts/common/fs/vnode.c

```diff
@@ -852,6 +852,31 @@ vn_rele(vnode_t *vp)
 	mutex_exit(&vp->v_lock);
 }
 
+void
+vn_phantom_rele(vnode_t *vp)
+{
+	VERIFY(vp->v_count > 0);
+
+	mutex_enter(&vp->v_lock);
+	VERIFY3U(vp->v_count, >=, vp->v_phantom_count);
+	vp->v_phantom_count--;
+	DTRACE_PROBE1(vn__phantom_rele, vnode_t *, vp);
+	if (vp->v_count == 1) {
+		mutex_exit(&vp->v_lock);
```

Comment:
CAN/SHOULD we `ASSERT0(vp->v_phantom_count);` prior to dropping the mutex?  IMHO it makes for nicer readability.

Reply:
Yeah, I can add that.